inspirehep/modules/editor/api.py

```diff
@@ -248,3 +250,34 @@ def get_workflow_and_schema(workflow_id):
         "workflow": workflow,
         "schema": load_schema(schema_url)
     })
+
+@blueprint_api.route('/validate_workflow', methods=['POST'])
+def validate_workflow():
+    workflow = request.json
+    data = workflow['metadata']
+
+    def get_validation_errors(data, schema):
+        """Creates a ``validation_errors`` dictionary.
+        Args:
+            data (dict): the object to validate.
+            schema (str): the name of the schema.
+        Returns:
+            dict: ``validation_errors`` formatted dict.
+        """
+        errors = _get_validation_errors(data, schema=schema)
+        error_messages = [
+            {
+                'path': map(text_type, error.absolute_path),
+                'message': text_type(error.message),
+            } for error in errors
+        ]
+        return error_messages
+
+    schema = load_schema('hep')
+
+    try:
+        validate(data, schema)
```

Comment:
you're validating twice, as get_validation_errors already does the validation: https://github.com/inspirehep/inspire-schemas/blob/7a077002941830a5b149b40f4601ebdd0c6c9792/inspire_schemas/utils.py#L777-L782. If `get_validation_errors` doesn't yield anything, it means the record is valid (you could test that to make sure).

Reply:
https://github.com/inspirehep/inspire-next/blob/d9ff04cbe726c3b84838ac384467658cb724a192/inspirehep/modules/workflows/utils/__init__.py#L452